Issue with NodeJS Codegen when no input and output types are generated

A simple provider like [pulumi-mysql](https://github.com/pulumi/pulumi-mysql) doesn't generate any input or output types.

In the generated `config/vars.ts`, there is an assumption that the input and output types exist:

```
import * as pulumi from "@pulumi/pulumi";
import * as inputs from "../types/input";
import * as outputs from "../types/output";
import * as utilities from "../utilities";
```

Therefore, we get a tsc compilation error:

```
yarn run v1.13.0
$ /Users/stack72/code/go/src/github.com/pulumi/pulumi-mysql/sdk/nodejs/node_modules/.bin/tsc
config/vars.ts:5:25 - error TS2307: Cannot find module '../types/input'.

5 import * as inputs from "../types/input";
                          ~~~~~~~~~~~~~~~~

config/vars.ts:6:26 - error TS2307: Cannot find module '../types/output'.

6 import * as outputs from "../types/output";
                           ~~~~~~~~~~~~~~~~~


Found 2 errors.
```

--- a/pkg/codegen/nodejs/gen.go
+++ b/pkg/codegen/nodejs/gen.go
@@ -918,7 +918,7 @@ func (mod *modContext) sdkImports(nested, utilities bool) []string {
 	rel, err := filepath.Rel(mod.mod, "")
 	contract.Assert(err == nil)
 	relRoot := path.Dir(filepath.ToSlash(rel))
-	if nested {
+	if nested && len(mod.types) > 0 {
 		imports = append(imports, fmt.Sprintf("import * as inputs from \"%s/types/input\";", relRoot))
 		imports = append(imports, fmt.Sprintf("import * as outputs from \"%s/types/output\";", relRoot))
 	}

--- a/pkg/codegen/nodejs/gen.go
+++ b/pkg/codegen/nodejs/gen.go
@@ -878,9 +878,9 @@ func (mod *modContext) configGetter(v *schema.Property) (string, string) {
 
 func (mod *modContext) genConfig(w io.Writer, variables []*schema.Property) error {
 	imports := map[string]codegen.StringSet{}
-	mod.getImports(variables, imports)
+	referencesNestedTypes := mod.getImports(variables, imports)
 
-	mod.genHeader(w, mod.sdkImports(true, true), imports)
+	mod.genHeader(w, mod.sdkImports(referencesNestedTypes, true), imports)
 
 	// Create a config bag for the variables to pull from.
 	fmt.Fprintf(w, "let __config = new pulumi.Config(\"%v\");\n", mod.pkg.Name)
@@ -918,7 +918,7 @@ func (mod *modContext) sdkImports(nested, utilities bool) []string {
 	rel, err := filepath.Rel(mod.mod, "")
 	contract.Assert(err == nil)
 	relRoot := path.Dir(filepath.ToSlash(rel))
-	if nested && len(mod.types) > 0 {
+	if nested {
 		imports = append(imports, fmt.Sprintf("import * as inputs from \"%s/types/input\";", relRoot))
 		imports = append(imports, fmt.Sprintf("import * as outputs from \"%s/types/output\";", relRoot))
 	}

